lib/kubernetes-deploy/restart_task.rb

```diff
@@ -27,58 +23,71 @@ def initialize(context:, namespace:, logger:)
       @context = context
       @namespace = namespace
       @logger = logger
-      @kubeclient = build_v1_kubeclient(context)
-      @v1beta1_kubeclient = build_v1beta1_kubeclient(context)
-      @policy_v1beta1_kubeclient = build_policy_v1beta1_kubeclient(context)
```

Comment:
I moved to initializing these lazily because they can raise. If the kubeconfig is invalid, I want to print a pretty error about it, not an exception (like deploys do).